jujugui/static/gui/src/app/components/entity-header/entity-header.js

```diff
@@ -55,7 +55,24 @@ YUI.add('entity-header', function() {
       @param {Object} e The click event
     */
     _handleDeployClick: function(e) {
-      this.props.deployService(this.props.entityModel);
+      var entityModel = this.props.entityModel;
+      var entity = entityModel.toEntity();
+      if (entity.type === 'charm') {
+        this.props.deployService(entityModel);
+        this._closeEntityDetails();
+      } else {
+        var id = entity.id.replace('cs:', '');
```

Comment:
I'd suggest to add some url resolving function in one place and use that in the code. If URLs change, it'll be hard to find all such examples. But not for this PR.